transaction_controller.go

```diff
@@ -0,0 +1,48 @@
+package amqp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/go-amqp/internal/encoding"
+)
+
+type TransactionControllerOptions struct {
+	// Capabilities is the list of extension capabilities the sender supports.
+	Capabilities []string
+}
+
+// TransactionController can interact with the transaction coordinator
+// for a transactional resource.
+// Reference: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#section-coordination
+type TransactionController struct {
+	sender *Sender
+}
+
+func (tc *TransactionController) Discharge(ctx context.Context, discharge TransactionDischarge, opts *SendOptions) error {
```

Comment:
This should take a `DischargeOptions` type so it can independently evolve.

Reply:
Done.